src/test/groovy/pl/allegro/tech/build/axion/release/infrastructure/git/GitRepositoryTest.groovy

```diff
@@ -220,6 +219,22 @@ class GitRepositoryTest extends Specification {
         remoteRawRepository.tag.list()*.fullName == ['refs/tags/release-push']
     }
 
+    def "should not push commits if the pushTagsOnly flag is set to true"() {
+        given:
+        initializationOptions = ScmInitializationOptions.fromProject(project, 'origin', true)
+        repository = new GitRepository(project.file('./repo'), identity, initializationOptions)
+
+        repository.tag('release-push')
+        repository.commit(['*'], 'commit after release-push')
+
+        when:
+        repository.push(ScmIdentity.defaultIdentity(), 'origin')
+
+        then:
+        remoteRawRepository.log(maxCommits: 1)*.fullMessage == ['InitialCommit']
```

Comment:
i'm bit confused and i don't know if this test tests the right thing - the new repository, created few lines up in `./repo` direcotry shouldn't have any remotes defined, only the "main" repository created in `setup` has this remote. Even if it somehow works as it should, for clarity i would like to have all definitions as explicit as possible, creating this repo from scratch would make the situation more understandable. Moving the flag into the `repository#push` would make this a lot easier :)